pkg/sources/git/git.go

```diff
@@ -417,7 +418,13 @@ func executeClone(ctx context.Context, params cloneParams) (*git.Repository, err
 		cloneURL.User = params.userInfo
 	}
 
-	gitArgs := []string{"clone", cloneURL.String(), params.clonePath}
+	gitArgs := []string{
+		"clone", cloneURL.String(), params.clonePath,
+		// Fetch additional refs from GitHub and GitLab.
+		// https://github.com/trufflesecurity/trufflehog/issues/1588
+		"-c", "remote.origin.fetch=+refs/pull/*:refs/heads/trufflehog/pull/*",
+		"-c", "remote.origin.fetch=+refs/merge-requests/*:refs/heads/trufflehog/merge-requests/*",
```

Comment:
This is super cool, but I don't actually understand how it works! How can these new refspecs find commits that aren't reachable from any refs (like is shown in your screenshot)?

Reply:
It's the `trufflehog` in the string that is throwing me off.. Was that a mistake? This comment did not mention it:

https://github.com/trufflesecurity/trufflehog/issues/1588#issuecomment-1706937906

Reply:
I'm also wondering what the behavior is for servers that might not support these options. I'm guessing it's ignored because GitLab uses 'merge request' terminology and GitHub uses 'pull request'

Reply:
@rgmz any thoughts on this?

Reply:
Git allows you to name refs whatever you'd like (e.g., `refs/notes`, `refs/foo/bar/baz`), [provided they follow some basic rules](https://git-scm.com/docs/git-check-ref-format#_description). However, the default configuration for `clone` / `fetch` doesn't pull all refs from a remote, only `refs/heads/*` and `refs/tags/*` (and maybe a few others, idk). 

These options instruct Git to pull any remote refs under `refs/{pull,merge-requests}/*` and map them locally to  `refs/heads/trufflehog/{pull,merge-requests}/*`. It has to be under `refs/heads/*` to be treated as a branch; the prefix of `trufflehog` was arbitrarily chosen to reduce the likelihood of branch name conflicts.